src/PhpUnit/JsonSchemaConstraint.php

```diff
@@ -21,25 +21,32 @@ class JsonSchemaConstraint extends Constraint
     private $context;
 
     /**
+     * @var Validator
+     */
+    private $validator;
+
+    /**
      * @param object $expectedSchema
      * @param string $context
+     * @param Validator $validator
      */
-    public function __construct($expectedSchema, $context)
+    public function __construct($expectedSchema, $context, Validator $validator)
     {
         parent::__construct();
 
         $this->expectedSchema = $expectedSchema;
         $this->context = $context;
+        $this->validator = $validator;
     }
 
     /**
      * {@inheritdoc}
      */
     protected function matches($other)
     {
-        $validator = $this->getValidator($other);
```

Comment:
Please add `$this->validator->reset();` follow a whiteline at the begin of this method.